esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -2193,15 +2193,15 @@ extern TPWRAPPER_API unsigned getContainerWUClusterInfo(CConstWUClusterInfoArray
     return clusters.ordinality();
 }
 
-extern TPWRAPPER_API IConstWUClusterInfo* queryContainerWUClusterInfoByName(const char* clustName, CConstWUClusterInfoArray& clusters)
+extern TPWRAPPER_API IConstWUClusterInfo* queryContainerWUClusterInfoByName(const char* clustName)
 {
-    ForEachItemIn(i, clusters)
+    Owned<IPropertyTreeIterator> queues = queryComponentConfig().getElements("queues");
+    ForEach(*queues)
```

Comment:
minor: we're case sensitive when dealing with these target queue names elsewhere from the config, so should e here too I think.
And so instead of iterating, could be xpath:
```
extern TPWRAPPER_API IConstWUClusterInfo* getContainerWUClusterInfoByName(const char* clustName)
{
    VStringBuffer xpath("queues/*[@name=\"%s\"], clusterName);
    IPropertyTree *queue = queryComponentConfig().queryPropTree(xpath);
    if (queue)
        return new CContainerWUClusterInfo(queue->queryProp("@name"), queue->queryProp("@type"), (unsigned) queue->getPropInt("@width", 1));
    return nullptr;
}
```

Reply:
Yes